src/main/java/uk/gov/pay/api/service/directdebit/DirectDebitPaymentSearchService.java

```diff
@@ -42,6 +68,39 @@ public Response doSearch(Account account, String reference, String state, String
         queryParams.put(PAGE, pageNumber);
         queryParams.put(DISPLAY_SIZE, displaySize);
 
-        return searchDirectDebitPayments.getSearchResponse(account, queryParams);
+        return getSearchResponse(account, queryParams);
+    }
+
+    private Response getSearchResponse(Account account, Map<String, String> queryParams) {
+        String url = directDebitConnectorUriGenerator.directDebitPaymentsURI(account, queryParams);
+        Response connectorResponse = client
+                .target(url)
+                .request()
+                .header(HttpHeaders.ACCEPT, APPLICATION_JSON)
+                .get();
+        LOGGER.info("response from dd connector for payment search: {}", connectorResponse);
+        if (connectorResponse.getStatus() == SC_OK) {
+            return processResponse(connectorResponse);
+        }
+        throw new SearchPaymentsException(connectorResponse);
+    }
+
+    private Response processResponse(Response directDebitResponse) {
+        try {
+            DirectDebitSearchResponse response = directDebitResponse.readEntity(DirectDebitSearchResponse.class);
+            List<DirectDebitPaymentForSearch> paymentFromResponse =
+                    response
+                            .getPayments()
+                            .stream()
+                            .map(payment -> DirectDebitPaymentForSearch.valueOf(
+                                    payment,
+                                    publicApiUriGenerator.getPaymentURI(payment.getPaymentId())
+                            )).collect(Collectors.toList());
+            HalRepresentation.HalRepresentationBuilder halRepresentation = HalRepresentation.builder()
+                    .addProperty("results", paymentFromResponse);
+            return Response.ok().entity(paginationDecorator.decoratePagination(halRepresentation, response, PAYMENT_PATH).build().toString()).build();
+        } catch (ProcessingException ex) {
+            throw new SearchPaymentsException(ex);
```

Comment:
see previous comments about catching ProcessingException

Reply:
can the catch be around the `connectorResponse.readEntity(DirectDebitSearchResponse.class)` please